fmlloader/src/main/java/net/minecraftforge/fml/loading/UniqueModListBuilder.java

```diff
@@ -75,6 +76,13 @@ public UniqueModListData buildUniqueList()
             final List<EarlyLoadingException.ExceptionData> duplicateModErrors = dupedMods.stream()
                                                                                    .map(dm -> new EarlyLoadingException.ExceptionData("fml.modloading.dupedmod", dm, Objects.toString(dm)))
                                                                                    .toList();
+            LOGGER.error(LOADING, "Found duplicate mods:\n{}", dupedMods.stream()
+                            .map(mod -> String.format("\tMod ID: '%s' from mod files: %s",
+                                    mod.getModId(),
+                                    modIds.get(mod.getModId()).stream()
+                                            .map(modInfo -> modInfo.getOwningFile().getFile().getFileName()).collect(joining(", "))
+                            )).collect(joining("\n")));
+
```

Comment:
Is there a reason this logic couldn't be included within the first stream when constructing the list?

Additionally, I'm wondering whether `fml.modloading.dupedmod` acts as a translation key in this context since we don't have anything defined for it.

Reply:
It could be included when the list is originally constructed, though it would make most sense with dupedMods changed to a map. 
I think `fml.modloading.dupedmod` is meant to be a translation key, but when I have actually encountered this error showing on screen (which is only in ancient versions, recently it never gets that far likely because of coremod or module errors) it has been raw and untranslated, and thus also devoid of any actual useful information, which makes sense if there is not anything defined for it.

Is it worth adding a lang entry for `fml.modloading.dupedmod` (and where is that done.. forges regular lang flle?), or should I change the thrown exception + exception data to just reuse the same format as I currently have?

Reply:
I would attempt to use `fml.modloading.dupedmod` with a lang entry first to see if it's manageable and sane since we already have the key implemented. It would be done in Forge's regular lang file afaik.

If that doesn't work or ends up being nonviable (probably due to language loading constraints or dynamic params), then just deprecate the key and replace with the format you have.

Reply:
I unable to test if the lang key works due to Forge crashing before any sort of window opens (because system mod discovery fails), so will go for the second option.

What do you mean by deprecating the lang key? Afaict this is the only use of it (as said before its not even defined in the lang file), so I don't see much point in not just straight removing it. 
I'm also not entirely sure how you go about deprecating it, cause its just a string which can't accept the normal annotation. Just a `/*Deprecated*/` comment?

Reply:
My bad. You would just remove it.